### PR TITLE
Fixed CodeDOM serialization of primitives

### PR DESCRIPTION
**Pull request by Tanya-Solyanik**
Serializer was adding unnecessary casts to primitive expressions.
For example ((int)(42)) instead of 42.

### CODE REVIEW

File: src/Common/tests/CommonTestHelper.cs [hunk 1]
```diff
@@ -451,17 +452,23 @@ public static TheoryData<EventArgs> GetEventArgsTheoryData()
             return data;
         }
 
-        public static TheoryData<Exception> GetSecurityOrCriticalException()
```
Comment (Tanya-Solyanik) [inline src/Common/tests/CommonTestHelper.cs]: Removed this function because it had not been used

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/PrimitiveCodeDomSerializer.cs [hunk 2]
```diff
@@ -49,14 +49,14 @@ public override object Serialize(IDesignerSerializationManager manager, object v
...
+                if (value is string stringValue)
                 {
-                    if (value is string stringValue && stringValue.Length > 200)
+                    if (stringValue.Length > 200)
                     {
                         expression = SerializeToResourceExpression(manager, stringValue);
                     }
                 }
-                else
+                else if (!(value is bool || value is char || value is int || value is float || value is double))
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/PrimitiveCodeDomSerializer.cs]: This test had been present in the original sources but vanished during the port :(

File: src/Common/tests/CommonTestHelper.cs [hunk 3]
```diff
@@ -163,12 +163,13 @@ internal static TheoryData<float> GetUFloatTheoryData()
         }
 
         // helper method to generate theory data for a span of string values
+        private const string reasonable = nameof(reasonable);
```
Comment (RussKie) [inline src/Common/tests/CommonTestHelper.cs]: [nit] missing blank line

File: src/System.Windows.Forms.Design/tests/UnitTests/Serialization/PrimitiveCodeDomSerializerTests.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        [Fact]
+        public void PrimitiveCodeDomSerializerTests_Constructor()
+        {
+            var underTest = new PrimitiveCodeDomSerializer();
+            Assert.NotNull(underTest);
+        }
+
+        [Theory]
+        [CommonMemberData(nameof(CommonTestHelper.GetPrimitiveExpressionTheoryData))]
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/tests/UnitTests/Serialization/PrimitiveCodeDomSerializerTests.cs]: I'm not a big fan of sharing test data as the data may change to suite a particular test case and break other tests. I've seen this in the past.
So if `GetPrimitiveExpressionTheoryData` isn't used anywhere else I suggest moving it into this class.

Also it is hard to review the test because the input data is in a separate file.
In this case, it would probably be simpler to rewrite as an `[InlineData]`:

```
[Theory]
[InlineData(false)]
[InlineData("aaa")]
[InlineData('a')]
[InlineData(42)]
[InlineData(42F)]
[InlineData(42.123)]
public void PrimitiveCodeDomSerializerTests_Serialize(object input)
```

Comment (RussKie) [reply src/System.Windows.Forms.Design/tests/UnitTests/Serialization/PrimitiveCodeDomSerializerTests.cs]: Do we need to test `null`?
Also a test for strings longer than 200 chars would be good

**Comment by RussKie** [comment]
A further possible refactor/cleanup is to invert `if` to reduce nesting and do early returns:
```
        public override object Serialize(IDesignerSerializationManager manager, object value)
        {
            using (TraceScope("PrimitiveCodeDomSerializer::" + nameof(Serialize)))
            {
                Trace("Value: {0}", (value == null ? "(null)" : value.ToString()));
            }

            CodeExpression expression = new CodePrimitiveExpression(value);

            if (value == null)
            {
                return expression;
            }

            if (value is string stringValue)
            {
                if (stringValue.Length > 200)
                {
                    return SerializeToResourceExpression(manager, stringValue);
                }

                return expression;
            }

            if (!(value is bool || value is char || value is int || value is float || value is double))
            {
                // Generate a cast for all other types because we won't parse them properly otherwise 
                // because we won't know to convert them to the narrow form.
                expression = new CodeCastExpression(new CodeTypeReference(value.GetType()), expression);
            }

            return expression;
        }
```

This way it is easier to follow the logic and write tests.

We can add a commit to this PR (happy to send it), or do it as a separate piece of work.

**Comment by RussKie** [comment]
Btw GitHub has a number of special keywords that help streamlining the issue management. You can find more info here - https://help.github.com/en/articles/closing-issues-using-keywords

For example, if a PR's commit message contains "Fixes #348" the underlying issue will be automatically closed one the PR is merged.
GitHub will also automatically link the issue with the PR and the commit.

File: src/System.Windows.Forms.Design/tests/UnitTests/Serialization/PrimitiveCodeDomSerializerTests.cs (hunk 4 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms.Design/tests/UnitTests/Serialization/PrimitiveCodeDomSerializerTests.cs]: long strings require additional mocking, at least, out of scope for this fix
